### PR TITLE
Issue 66/secondary partner styling enhancements

### LINKED ISSUE #66: Secondary Partner Styling Enhacements
### Describe the Current Behavior/Feature:

- Reduce padding between text and Author
- Try to reduce card height while keeping them all symmetrical 

---

### Proposed Implementation:

Ian's strong options on testimonial Author
- Remove dash
- Bold author
- Center author

### PR DESCRIPTION
**Pull request by andycwilliams**
## This PR:

Resolves #66

A relatively quick go at it. Very possibly some minor adjustments left to make but it should hopefully address all of the issue's core needs.

---

## Screenshots (if applicable):

---

## Future Steps/PRs Needed to Finish This Work (optional):

Fine-tune layout as needed.

### CODE REVIEW

File: src/components/global/Footer.jsx [hunk 1]
```diff
@@ -173,7 +173,7 @@ const Footer = () => {
         <Box
           component="img"
           alt="CODE PDX logo"
-          aria-label="Code pdx"
+          aria-label="CODE PDX"
```
Comment (xscottxbrownx) [inline src/components/global/Footer.jsx]: not sure it needs an aria-label if it has an alt, but glad to change this for continuityl

File: src/components/home/partners/SecondaryPartners.jsx [hunk 2]
```diff
@@ -65,25 +67,29 @@ const LargeScreenComponent = ({
...
+        <Typography variant="caption" sx={{ fontWeight: 'bold' }}>
           {testimonialAuthor}
         </Typography>
         {testimonialTwo ? (
-          <Typography variant="body1" display={'flex'} p={'40px 15% 5% 0'}>
+          <Typography
+            variant="body1"
+            // Below not necessary?
+            // display={'flex'}
+            p={'40px 15% 5% 0'}
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: Any reason this wouldn't have the same padding as `testimonial`? the text doesn't line up for testimonial one and two due to this.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 3]
```diff
@@ -130,16 +136,29 @@ const SmallScreenComponent = ({
...
+          variant="caption"
+          p={'0 5% 8% 5%'}
+          sx={{ fontWeight: 'bold', textAlign: 'center' }}
+        >
           {testimonialAuthor}
         </Typography>
         {testimonialTwo ? (
-          <Typography variant="body1" display={'flex'} p={'5% 5% 5% 5%'}>
+          <Typography
+            variant="body1"
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: not necessary. the `Typography` MUI component does not have a `display` attribute. This does nothing.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 4]
```diff
@@ -130,16 +136,29 @@ const SmallScreenComponent = ({
...
+            p={'5% 5% 5% 5%'}
+          >
             {testimonialTwo}
           </Typography>
         ) : null}
         {testimonialAuthorTwo ? (
-          <Typography variant="caption" p={'0 5% 10% 5%'}>
+          <Typography
+            variant="caption"
+            p={'0 5% 10% 5%'}
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: any reason for this padding not to be the same as `testimonial Author`?

File: src/components/home/partners/SecondaryPartners.jsx [hunk 5]
```diff
@@ -155,6 +174,8 @@ const SecondaryPartners = () => {
   const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));
 
   return (
+    // This Container element causes the second partners to be narrower than the primary partner, even on small screens.
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: I would personally prefer them to be the same size horizontally. They are already a different layout and have less info than the primary partner. I think that's enough to delineate the two.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 6]
```diff
@@ -1,21 +1,22 @@
+// Material UI Imports
 import Box from '@mui/material/Box';
 import Container from '@mui/material/Container';
 import Grid from '@mui/material/Grid';
 import Typography from '@mui/material/Typography';
-
 import useMediaQuery from '@mui/material/useMediaQuery';
 import { useTheme } from '@mui/material/styles';
+// Data Imports
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: here you used "Data Imports", in `VolunteerSteps` you used "Component Imports", in `ProjectsBrief` it's "Data Imports"...

so maybe change it in `VolunteerSteps` to match?  It honestly doesn't matter, but if we're going through and changing for consistency, then that one too.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 7]
```diff
@@ -1,21 +1,22 @@
...
 import { secondaryPartnerList } from './secondaryPartnerList';
-
+// PropTypes Imports
 import { PropTypes } from 'prop-types';
 
 const secondaryPartnerProps = {
   index: PropTypes.number.isRequired,
   company: PropTypes.string.isRequired,
   testimonial: PropTypes.string.isRequired,
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: If we are moving these around, I'd prefer 
```
testimonial
testimonialAuthor
testimonialTwo
testimonialAuthorTwo
```
but would actually prefer `testimonialAuthorTwo` to be changed in the codebase to `testimonialTwoAuthor` 😂

File: src/components/home/partners/SecondaryPartners.jsx [hunk 8]
```diff
@@ -52,9 +53,10 @@ const LargeScreenComponent = ({
       sx={{
         background: gradientStyle,
         borderRadius: '30px',
-        minHeight: '500px',
-        marginBottom: '100px',
-        flexDirection: 'row'
+        minHeight: '350px',
+        marginBottom: '100px'
+        // Below not necessary?
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: Nope not necessary.  Default for `display: flex` is row already...

MUI `<Grid>` is based on flexbox

Screenshot 2023-12-19 at 8 59 50 AM

File: src/components/home/partners/SecondaryPartners.jsx [hunk 9]
```diff
@@ -65,25 +67,29 @@ const LargeScreenComponent = ({
...
-        <Typography variant="body1" p={'5% 15% 3% 0'}>
+      <Grid item pl={'6%'} pr={'1%'} {...partnerGridStyle} order={contentOrder}>
+        <Typography variant="body1" p={'5% 0 3% 0'}>
           {testimonial}
         </Typography>
-        <Typography variant="caption" p={'15px 0 5% 0'}>
+        <Typography variant="caption" sx={{ fontWeight: 'bold' }}>
           {testimonialAuthor}
         </Typography>
         {testimonialTwo ? (
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: on both `LargeScreenComponent` and `SmallScreenComponent` there are ternaries used that are just something to the effect of:

```js
{condition ? (
    <Returned JSX>
) : null}
```

if it's "this" or "null", I think it's more commonplace in React (and I'd personally prefer) to use the JS AND operand short-circuiting instead:

```js
{condition && (
    <Returned JSX>
)}
```

if condition is true, the right of AND gets evaluated and rendered.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 10]
```diff
@@ -65,25 +67,29 @@ const LargeScreenComponent = ({
...
-          <Typography variant="body1" display={'flex'} p={'40px 15% 5% 0'}>
+          <Typography
+            variant="body1"
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: not necessary. the `Typography` MUI component does not have a `display` attribute. This does nothing.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 11]
```diff
@@ -65,25 +67,29 @@ const LargeScreenComponent = ({
             alt={`${company} logo`}
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: This `<Box>` could be self-closing instead.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 12]
```diff
@@ -65,25 +67,29 @@ const LargeScreenComponent = ({
...
             aria-label={`${company} logo`}
             src={partnerLogo}
-            mb={'150px'}
             width={'250px'}
           ></Box>
         </a>
       </Grid>
-      <Grid item pl={'6%'} {...partnerGridStyle} order={contentOrder}>
-        <Typography variant="body1" p={'5% 15% 3% 0'}>
+      <Grid item pl={'6%'} pr={'1%'} {...partnerGridStyle} order={contentOrder}>
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: ### This whole section on, I'm not sure about this current approach...

There's just so much inconsistency with all the different paddings on different things. It's causing so the text isn't lined up with the logos vertically (when in desktop), the authors have different paddings, etc.

Like visually (besides the vertical alignment) looks like it does the job, but under the hood as a dev - it seems like plug in whatever wherever to get there.

---

I feel like the left `Grid item` (company logo on first `secondaryPartner`) should be vertically and horizontally centered.

Then the right `Grid item` should be the same, but that encompasses all the testimonials and authors it has. This `Grid item` or the `Grid container` itself should have the padding/margin applied for the styling properly.

Then within the right `Grid item` make the spacing between elements as see fit, but should just be `py` or `pt` and `pb` at that point.  Or even better use a `Stack` and `spacing={whatever}` so it's all consistent and easy to determine what the expected behavior/outcome is. Future maintainability.

**Review by xscottxbrownx — CHANGES_REQUESTED** [review]

File: src/components/global/Footer.jsx (hunk 1 above)
Comment (andycwilliams) [reply src/components/global/Footer.jsx]: Missed that. I was searching through the code and just updating the name wherever it appeared.

File: src/components/home/partners/SecondaryPartners.jsx (hunk 2 above)
Comment (andycwilliams) [reply src/components/home/partners/SecondaryPartners.jsx]: Fixed

File: src/components/home/partners/SecondaryPartners.jsx (hunk 4 above)
Comment (andycwilliams) [reply src/components/home/partners/SecondaryPartners.jsx]: For some reason it looks less consistent, at least to me. But I can change it.

File: src/components/home/partners/SecondaryPartners.jsx (hunk 5 above)
Comment (andycwilliams) [reply src/components/home/partners/SecondaryPartners.jsx]: That's my thinking too.

File: src/components/home/partners/SecondaryPartners.jsx [hunk 13]
```diff
@@ -130,19 +128,27 @@ const SmallScreenComponent = ({
...
+          <Typography variant="body1" p={'5% 5% 5% 5%'}>
             {testimonialTwo}
           </Typography>
-        ) : null}
-        {testimonialAuthorTwo ? (
-          <Typography variant="caption" p={'0 5% 10% 5%'}>
+        )}
+        {testimonialAuthorTwo && (
+          <Typography
+            variant="caption"
```
Comment (xscottxbrownx) [inline src/components/home/partners/SecondaryPartners.jsx]: Think the padding for `testimonialAuthorTwo` should be `0 5% 8% 5%` to match `testimonialAuthor`

Believe line 146 is reason why there is more space between the second testimonial and it's author vs the first.
Screenshot 2023-12-19 at 4 57 17 PM
Screenshot 2023-12-19 at 4 57 43 PM

**Review by xscottxbrownx — COMMENTED** [review]
see the photos for what I was trying to explain with padding earlier

File: src/components/home/partners/SecondaryPartners.jsx (hunk 4 above)
Comment (xscottxbrownx) [reply src/components/home/partners/SecondaryPartners.jsx]: see photos below

**Review by Jared-Krajewski — CHANGES_REQUESTED** [review]
padding seems to still be at previous values once viewports get smaller. otherwise good enough for me once this & Scott's requests are fulfilled.

https://github.com/codeforpdx/codepdx_website/assets/114430929/60226884-9bec-4a31-94dc-f568ee04343c

**Comment by andycwilliams** [comment]
It should be more consistent along viewports, though the code itself is kind of janky (padding measured by pixels as well as percentages). More so than I'd like but this is what I have time for at the moment.

Let me know if this is good enough. I'll have more time to tweak it later on.

**Comment by Jared-Krajewski** [comment]
> It should be more consistent along viewports, though the code itself is kind of janky (padding measured by pixels as well as percentages). More so than I'd like but this is what I have time for at the moment.
> 
> Let me know if this is good enough. I'll have more time to tweak it later on.
Screenshot 2023-12-27 at 1 47 03 PM


https://github.com/codeforpdx/codepdx_website/assets/114430929/043cacd7-aa99-4597-98fb-c16e43613c32



the card heights and width are all quite different now and weren't supposed to be changed as part of this. Those need to be reverted to previous values, the goal was to trim some of the height from the ORDSN card by widening its the text areas within the current box, remove some of the extra padding from authors and match the rest of the cards to whatever height that ended up being. Let me know if you have time to fix this if not I can try to finish it when I have time.

**Comment by andycwilliams** [comment]
Still not finished. Need to center the author text, doesn't look great on smaller viewports, etc. But I don't have too much more time for this one, since I still have a couple other issues to do. @Jared-Krajewski can take over if you want.

I took a swing at refactoring it so that there wouldn't need to be quite so much padding in so many areas. Can change it back if you don't like. I added `Box`es to the grid items and imo it's better but still needs fine-tuning.

There are some annoying things, like the `Container` on line 155 makes it more consistent with `PrimaryPartner` on larger screens but less so on smaller ones. 

The `LargeScreenComponent` and `SmallScreenComponent` make it slightly more awkward to deal with, since this is one of the only places we use media queries. Otherwise we use the breakpoints that Material UI comes with. But I didn't want to just refactor the entire thing.

**Comment by xscottxbrownx** [comment]
pushed fix for centering testimonial authors... switched `caption` variant to `subtitle2` for quick fix.   Other option would've been to adjust the theme of the `caption` variant as that was the issue.

**Comment by xscottxbrownx** [comment]
> The `LargeScreenComponent` and `SmallScreenComponent` make it slightly more awkward to deal with, since this is one of the only places we use media queries. Otherwise we use the breakpoints that Material UI comes with. But I didn't want to just refactor the entire thing.

If you don't like the two separate components, I refactored into just a single component with the styling conditionals built in. Not sure if it's cleaner/better, but maybe 🤷🏻‍♂️

```js
// Material UI Imports
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';
import useMediaQuery from '@mui/material/useMediaQuery';
import { useTheme } from '@mui/material/styles';
// Other Library Imports
import { PropTypes } from 'prop-types';
// Component Imports
import { secondaryPartnerList } from './secondaryPartnerList';


const partnerGridStyle = {
  xs: 12,
  md: 6,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  flexDirection: 'column',
  p: '4%'
};

const calculateGradientStyle = (isSmallScreen, index) => {
  if (isSmallScreen) {
    return 'linear-gradient(180deg, rgba(217, 217, 217, 0) 38.54%, rgba(217, 217, 217, 0.4) 82.29%)';
  }

  return index % 2 === 0
    ? 'linear-gradient(90deg, rgba(217, 217, 217, 0) 38.54%, rgba(217, 217, 217, 0.4) 82.29%)'
    : 'linear-gradient(270deg, rgba(217, 217, 217, 0) 38.54%, rgba(217, 217, 217, 0.4) 82.29%)';
};

const SecondaryPartners2 = () => {
  const theme = useTheme();
  const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));

  return (
    <Container>

      {secondaryPartnerList.map((partner, index) => {
        const gradientStyle = calculateGradientStyle(isSmallScreen, index);
        return (
          <Grid
            key={partner.company}
            container
            sx={{
              background: gradientStyle,
              borderRadius: '30px',
              minHeight: isSmallScreen ? 'auto' : '350px',
              marginBottom: isSmallScreen ? '15%' : '100px'
            }}
          >
            <Grid item {...partnerGridStyle} order={isSmallScreen ? 0 : index % 2}>
              <a href={partner.website} target="_blank" rel="noopener noreferrer">
                <Box
                  component="img"
                  alt={`${partner.company} logo`}
                  aria-label={`${partner.company} logo`}
                  src={partner.partnerLogo}
                  width={isSmallScreen ? '100px' : '250px'}
                />
              </a>
            </Grid>
            <Grid item {...partnerGridStyle} order={isSmallScreen ? 1 : (index + 1) % 2}>
              <Stack spacing={1}>
                <Typography variant="body1">{partner.testimonial}</Typography>
                <Typography
                  variant="subtitle2"
                  sx={{
                    fontWeight: 'bold',
                    textAlign: 'center'
                  }}
                >
                  {partner.testimonialAuthor}
                </Typography>
              </Stack>
              <br />
              <Stack spacing={1}>
                {partner.testimonialTwo && (
                  <Typography variant="body1">{partner.testimonialTwo}</Typography>
                )}
                {partner.testimonialTwoAuthor && (
                  <Typography variant="subtitle2" sx={{ fontWeight: 'bold', textAlign: 'center' }}>
                    {partner.testimonialTwoAuthor}
                  </Typography>
                )}
              </Stack>
            </Grid>
          </Grid>
        );
      })}

    </Container>
  );
};

SecondaryPartners2.propTypes = {
  index: PropTypes.number.isRequired,
  company: PropTypes.string.isRequired,
  testimonial: PropTypes.string.isRequired,
  testimonialAuthor: PropTypes.string,
  testimonialTwo: PropTypes.string,
  testimonialTwoAuthor: PropTypes.string,
  partnerLogo: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired
};

export default SecondaryPartners2;
```

`partner` in the .map() could be destructured if wanted.

**Comment by andycwilliams** [comment]
Is there anything else needed from this?

**Review by Jared-Krajewski — APPROVED** [review]
Good enough for now. we can touch back on this another time if we want to update anything.

**Review by xscottxbrownx — APPROVED** [review]

**Pull request merged**